src/main/java/com/kalessil/phpStorm/phpInspectionsEA/options/OptionsComponent.java

```diff
@@ -0,0 +1,163 @@
+package com.kalessil.phpStorm.phpInspectionsEA.options;
+
+import com.kalessil.phpStorm.phpInspectionsEA.gui.PrettyListControl;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Helps on constructions of a normalized inspection options list for IntelliJ's IDE.
+ */
+public class OptionsComponent {
+    /**
+     * Stores the options panel.
+     */
+    private final JPanel optionsPanel;
+
+    /**
+     * Construct the OptionsComponent.
+     */
+    private OptionsComponent() {
+        optionsPanel = new JPanel();
+        optionsPanel.setLayout(new MigLayout());
+    }
+
+    /**
+     * Creates a new OptionsComponent.
+     *
+     * @param componentDefiner The component definer, that will create the option fields.
+     */
+    static public JPanel create(@NotNull final Consumer<OptionsComponent> componentDefiner) {
+        // We should create a OptionsComponent that offer the features to create fields.
+        // Then the component definer will use that new instance to create them.
+        final OptionsComponent component = new OptionsComponent();
+        componentDefiner.accept(component);
+
+        // Returns only the JPanel component.
+        return component.optionsPanel;
+    }
+
+    /**
+     * Creates a Checkbox.
+     *
+     * @param fieldTitle     The title of the field on options panel.
+     * @param defaultValue   The default or current field value stored on inspection.
+     * @param updateConsumer The update consumer (called after item change).
+     */
+    public JCheckBox createCheckbox(final String fieldTitle, final boolean defaultValue, final Consumer<Boolean> updateConsumer) {
+        final JCheckBox createdCheckbox = new JCheckBox(fieldTitle, defaultValue);
+        createdCheckbox.addItemListener((itemEvent) -> updateConsumer.accept(createdCheckbox.isSelected()));
+
+        optionsPanel.add(createdCheckbox, "wrap");
+
+        return createdCheckbox;
+    }
+
+    /**
+     * Creates a new Radio component.
+     *
+     * @param componentDefiner The component definer, that will create the option radios.
+     */
+    public void createRadio(@NotNull final Consumer<RadioComponent> componentDefiner) {
+        final RadioComponent radioComponent = new RadioComponent();
+        componentDefiner.accept(radioComponent);
+    }
+
+    /**
+     * Creates a List.
+     *
+     * @param fieldTitle  The title of this field on panel.
+     * @param listItems   The default or current list items.
```

Comment:
Initially, I tried to use the `AtomicBoolean` that initially receives the "default value" defined on field of inspection class. Then for all components I create this parameter ("default or current"). Next, I found that it is impossible currently, because IDE don't recognize the `AtomicBoolean` as a configurable option (_it just ignore it_).

The unique type that we can have as a _non-primitive type_ is a `List<>`. Then it could have the default value (which is defined from inspection class) or the current value (after IDE load that). It are "magic fields" which are strangely managed by IDE (rather than doing it more efficiently).

To summarize: this variable is used both to read the default values and to read the current configuration.